docs/sdk/client-side-sdks/flutter.md

```diff
@@ -0,0 +1,295 @@
+
+```dart
+if (boolVariable.value == true) {
+    // Run Feature Flag Code
+} else {
+    // Run Default Code
+}
+```
+
+The `Variable` object also contains the following params: 
+    - `key`: the key indentifier for the Variable
+    - `type`: the type of the Variable, one of: `String` / `Boolean` / `Number` / `JSON`
+    - `value`: the Variable's value
+    - `defaultValue`: the Variable's default value
+    - `isDefaulted`: if the Variable is using the `defaultValue`
+    - `evalReason`: evaluation reason for why the variable was bucketed into its value
+
+If the value is not ready, it will return the default value passed in the creation of the variable.
+
+### Variable Updates
+
+Variable values update whenever `identifyUser()` or `resetUser()` are called, or when the project configuration changes (to learn more, visit our [Realtime Updates](/docs/sdk/features/realtime-updates) page).
+
+To listen for variable updates, the `onUpdate()` method can be used. Please note, a strong reference to the variable is needed for `onUpdate` to be triggered.
+
+```dart
+final variable = await _dvcClient.variable('my-variable', 'Default Value');
+variable?.onUpdate((updatedVariable) {
+        // Variable value updated updatedVariable.value
+});
+```
+
+### Getting All Features / Variables
+
+### Get All Features
+
+To get all the Features returned in the config:
+
+```dart
+Map<String, DVCFeature> features = await _dvcClient.allFeatures();
+```
+
+If the SDK has not finished initializing, these methods will return an empty object.
+
+
+### Get All Variables
+
+To get all the variables returned in the config:
+
+```dart
+Map<String, DVCVariable> variables = await _dvcClient.allVariables();
+```
+
+If the SDK has not finished initializing, these methods will return an empty object.
+
+### Identifying User
+
+To identify a different user, or the same user passed into the initialize method with more attributes, 
+build a DVCUser object and pass it into `identifyUser`:
+
+```dart
+DVCUser user = DVCUserBuilder()
+    .userId('my-user1')
+    .email('my-email@email.com')
+    .country("CA")
+    .name("My Name")
+    .language("EN")
+    .customData({ "customkey": "customValue" })
+    .privateCustomData({ "customkey2": "customValue2" })
+    .build()
+
+_dvcClient.identifyUser(user)
+```
+
+To wait on Variables that will be returned from the identify call, you can pass in a DVCCallback:
+
+```dart
+_dvcClient.identifyUser(user, (error, variables) => {
+    // Error or Variables for the Identified User
+})
+```
+
+If `error` exists the called the user's configuration will not be updated and previous user's data will persist.
+
+### Reset User
+
+To reset the user into an anonymous user, `resetUser` will reset to the anonymous user created before 
+or will create one with an anonymous `user_id`.
+
+```dart
+_dvcClient.resetUser();
+```
+
+To wait on the Features of the anonymous user, you can pass in a DVCCallback:
```

Comment:
We don't define the type DVCCallback, should we just refer to this as a callback?